drivers/SmartThings/zigbee-window-treatment/src/mic/init.lua

```diff
@@ -0,0 +1,187 @@
+
+    local invert_value = device.preferences.reverse
+    local invert_cluster_cmd = cluster_base.write_manufacturer_specific_attribute(device, INVERT_CLUSTER,
+        INVERT_CLUSTER_ATTRIBUTE, 0x0000, data_types.Boolean, invert_value)
+    device:send(invert_cluster_cmd)
+end
+
+local function info_changed(driver, device, event, args)
+    if device.preferences ~= nil and device.preferences.reverse ~= args.old_st_store.preferences.reverse then
+        invert_preference_handler(device)
+    end
+end
+
+local set_window_shade_level = function(level)
+
+    return function(driver, device, cmd)
+        device:send_to_component(cmd.component, WindowCovering.server.commands.GoToLiftPercentage(device, level))
+    end
+end
+
+local WindowCovering = zcl_clusters.WindowCovering
+
+local SHADE_SET_STATUS = "shade_set_status"
+
+local function current_position_attr_handler(driver, device, value, zb_rx)
+
+    local level = value.value
+    local current_level = device:get_latest_state("main", capabilities.windowShadeLevel.ID,
+        capabilities.windowShadeLevel.shadeLevel.NAME)
+    local windowShade = capabilities.windowShade.windowShade
+
+    if level == 0 then
+        device:emit_event(windowShade.closed())
+        device:emit_event(capabilities.windowShadeLevel.shadeLevel(0))
+    elseif level == 100 then
+        device:emit_event(windowShade.open())
+        device:emit_event(capabilities.windowShadeLevel.shadeLevel(100))
+    else
+        if current_level ~= level or current_level == nil then
+            current_level = current_level or 0
+            device:emit_event(capabilities.windowShadeLevel.shadeLevel(level))
+            local event = nil
+
+            if current_level ~= level then
+                --  event = current_level < level and windowShade.opening() or windowShade.closing()
+                event = current_level < level and windowShade.partially_open() or windowShade.partially_open()
+            end
+            if event ~= nil then
+                device:emit_event(event)
+            end
+        end
+
+        local set_status_timer = device:get_field(SHADE_SET_STATUS)
+        if set_status_timer then
+            device.thread:cancel_timer(set_status_timer)
+            device:set_field(SHADE_SET_STATUS, nil)
+        end
+        local set_window_shade_status = function()
+            local current_level = device:get_latest_state("main", capabilities.windowShadeLevel.ID,
+                capabilities.windowShadeLevel.shadeLevel.NAME)
+
+            if current_level == 0 then
+                device:emit_event(windowShade.closed())
+            elseif current_level == 100 then
+                device:emit_event(windowShade.open())
+            else
+                device:emit_event(windowShade.partially_open())
+            end
+        end
+
+        set_status_timer = device.thread:call_with_delay(1, set_window_shade_status)
+        device:set_field(SHADE_SET_STATUS, set_status_timer)
```

Comment:
this timer stuff from the default is only there because we output intermediate `opening` and `closing` states. If you are always just going to emit `partially_open` then you don't need to do this